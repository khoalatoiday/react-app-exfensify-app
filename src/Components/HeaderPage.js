import { NavLink } from "react-router-dom";
import React from "react";
const HeaderPage = () => (
  <header>
    <h1>Exfensify app</h1>
    <li>
      <NavLink to="/help" activeClassName="is-active">
        Help Page
      </NavLink>
    </li>
    <li>
      <NavLink to="/create" activeClassName="is-active">
        Create Page
      </NavLink>
    </li>
    <li>
      <NavLink to="/" activeClassName="is-active" exact={true}>
        Home Page
      </NavLink>{" "}
    </li>
  </header>
);

export { HeaderPage as default };
