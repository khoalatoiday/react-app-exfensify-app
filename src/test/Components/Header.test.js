/*
    - Sử dụng các API của enzyme cùng toMatchSnapshot của JEST để test RC Component
    - shallow được sử dụng để test JSX code
    - JEST sẽ tạo một snapshot và lưu result được tạo bởi shallow vào snapshot với expect(result).toMatchSnapshot(). Nếu 
    nó được chạy lần đầu thì nó sẽ tạo mới snapshot. Trong trường hợp đã tạo rồi thì khi chạy toMatchSnapshot() thì hệ thống 
    sẽ so sánh result với snapshot đã tồn tại
    - Nếu muốn cập nhật snapshot thì chọn "u"
*/

import React from "react";
import HeaderPage from "../../Components/HeaderPage";
import { shallow } from "enzyme";
// import toJSON from "enzyme-to-json" // set up "snapshotSerializers" tại jest.config.json thì không cần phải gọi API liên tục nữa
test("Should render Header component correctly", () => {
  const wrapper = shallow(<HeaderPage />);
//   expect(toJSON(wrapper)).toMatchSnapshot();
  expect(wrapper).toMatchSnapshot();
});
