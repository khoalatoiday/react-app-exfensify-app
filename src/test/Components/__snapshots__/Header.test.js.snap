// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header component correctly 1`] = `
<header>
  <h1>
    Exfensify app
  </h1>
  <li>
    <NavLink
      activeClassName="is-active"
      to="/help"
    >
      Help Page
    </NavLink>
  </li>
  <li>
    <NavLink
      activeClassName="is-active"
      to="/create"
    >
      Create Page
    </NavLink>
  </li>
  <li>
    <NavLink
      activeClassName="is-active"
      exact={true}
      to="/"
    >
      Home Page
    </NavLink>
     
  </li>
</header>
`;
